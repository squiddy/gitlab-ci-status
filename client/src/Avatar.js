import React from "react";

export function Avatar({ obj, className = "" }) {
  // Renders the avatar of either a user or project, if available. Falls back to
  // a svg placeholder otherwise.
  const imageBaseUrl = "http://gitlab.bof.mm.local";

  if (obj && obj.avatar_url) {
    // Handle both relative (gitlab) or absolute (e.g. gravatar) URLs.
    const isAbsolute = obj.avatar_url.startsWith("http");

    const finalUrl = isAbsolute
      ? obj.avatar_url
      : imageBaseUrl + obj.avatar_url;
    return <img className={`avatar ${className}`} src={finalUrl} alt="" />;
  } else {
    return (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="14"
        height="14"
        viewBox="0 0 14 14"
        className={`avatar avatar-empty ${className}`}
      >
        <circle cx="7" cy="7" r="6" />
      </svg>
    );
  }
}
