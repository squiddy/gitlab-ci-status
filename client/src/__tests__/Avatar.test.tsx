import React from 'react';
import renderer from 'react-test-renderer';

import { Avatar } from '../Avatar';

it('Avatar renders empty when no data is available', () => {
  const tree = renderer.create(<Avatar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Avatar renders correctly with relative image URL', () => {
  const tree = renderer
    .create(<Avatar obj={{ avatar_url: '/kitten.jpg' }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Avatar renders correctly with absolute image URL', () => {
  const tree = renderer
    .create(<Avatar obj={{ avatar_url: 'http://example.com/kitten.jpg' }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
