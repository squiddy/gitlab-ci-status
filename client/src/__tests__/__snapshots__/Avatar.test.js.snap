// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with absolute image URL 1`] = `
<img
  alt=""
  className="avatar"
  src="http://example.com/kitten.jpg"
/>
`;

exports[`Avatar renders correctly with relative image URL 1`] = `
<img
  alt=""
  className="avatar"
  src="http://gitlab.bof.mm.local/kitten.jpg"
/>
`;

exports[`Avatar renders empty when no data is available 1`] = `
<svg
  className="avatar avatar-empty"
  height="14"
  viewBox="0 0 14 14"
  width="14"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    cx="7"
    cy="7"
    r="6"
  />
</svg>
`;
